airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams.py

```diff
@@ -206,11 +223,11 @@ def read_records(
         stream_state: Mapping[str, Any] = None,
     ) -> Iterable[Mapping[str, Any]]:
         """Read records using batch API"""
-        records = self._read_records(params=self.request_params(stream_state=stream_state))
-        requests = [record.api_get(fields=self.fields, pending=True) for record in records]
-        for requests_batch in batch(requests, size=self.batch_size):
-            for record in self.execute_in_batch(requests_batch):
-                yield self.clear_urls(record)
+        for records in self._read_records(params=self.request_params(stream_state=stream_state)):
+            requests = [record.api_get(fields=self.fields, pending=True) for record in records]
+            for requests_batch in batch(requests, size=self.batch_size):
+                for record in self.execute_in_batch(requests_batch):
+                    yield self.clear_urls(record)
```

Comment:
what is the motivation behind these changes?

Reply:
When syncing using the "all accounts" strategy, we could have hundreds of accounts to sync (it is our case, 500+), using this approach, we fetch and treat records one account at a time and leverage the `@backoff_policy` on `_read_records()`.

Reply:
I mean the only change I see here is an indentation, so I was curious

Reply:
Makes sense, to get a better picture, also check what changed in the different `def _read_records()` methods. If you configure a single account, the outcome is the same. If you configure multiple accounts, this change batch processes one account at a time.